app/lib/proxy_api/available_proxy.rb

```diff
@@ -0,0 +1,28 @@
+module ::ProxyAPI
+  class AvailableProxy < ::ProxyAPI::Resource
+
+    HTTP_ERRORS = [
+        EOFError,
+        Errno::ECONNRESET,
+        Errno::EINVAL,
+        Net::HTTPBadResponse,
+        Net::HTTPHeaderSyntaxError,
+        Net::ProtocolError,
+        Timeout::Error
+    ]
+
+    def initialize(args)
+      @url = args[:url]
+      super args
+    end
+
+    def available?
```

Comment:
we should migrate to smart_proxy_dynflow on smart proxy side so we could use its API to check proxy load, but seems good enough for now